### PR TITLE
[stdloc] add locator with focus on local seismicity

### PR DESCRIPTION
**Pull request by luca-s**
As discussed #83 this PR provides the inclusion of a new locator.  You can have a look at the current implementation and suggest any improvements or changes you would like to have before the integration. 

I have tested the code with local seismicity, but also at regional scale. I can see the results are pretty consistent with LOCSAT. I didn't write the locator with regional seismicity in mind, but the method is general enough so it works anyway and it makes testing easier.

One feature that I would like to add is the computation of the uncertainty ellipsoid. Hopefully I will find the time to add it soon.

### CODE REVIEW

File: plugins/locator/genloc/genloc.cpp [hunk 1]
```diff
@@ -0,0 +1,1331 @@
...
+using IDList = Seiscomp::Seismology::LocatorInterface::IDList;
+using LocatorException = Seiscomp::Seismology::LocatorException;
+using StationNotFoundException = Seiscomp::Seismology::StationNotFoundException;
+using PickNotFoundException = Seiscomp::Seismology::PickNotFoundException;
+
+namespace {
+// <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+
+// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+ADD_SC_PLUGIN("Generic Locator", "gempa GmbH <seiscomp-devel@gempa.de>", 1, 0,
```
Comment (gempa-jabe) [inline plugins/locator/genloc/genloc.cpp:57]: Please add yourself as author here.

**Comment by gempa-jabe** [comment]
Just to avoid misunderstandings, we are not talking about a port of https://www.sciencedirect.com/science/article/pii/S0098300404001438 to SeisComP, right? Both projects just share the name by coincidence?

Otherwise it might be helpful to include a reference into the documentation or add a note that it is not to be confused with GENLOC.

**Comment by luca-s** [comment]
> Just to avoid misunderstandings, we are not talking about a port of https://www.sciencedirect.com/science/article/pii/S0098300404001438 to SeisComP, right? Both projects just share the name by coincidence?
 
Just a coincidence.

**Comment by luca-s** [comment]
@gempa-jabe I am now reading the article you linked. The similarities are impressive and I am tempted to change name

**Comment by gempa-jabe** [comment]
Maybe we are talking about the similar method published in another paper unless you came up with that locator in a dark room by yourself. ;)

The GENLOC method is implemented in a commercial module (not ours) and maybe that could cause conflicts. So renaming it might be wise.

**Comment by luca-s** [comment]
Well, the methods are basic and well known, that's why we both come up with the name gen(eric)loc. However I used, as a reference, the book "Routine Data Processing in Earthquake Seismology". 

I should have checked the name was not already in use. I will rename this locator.

**Comment by luca-s** [comment]
For now let's call it `stdloc`: standard method locator. I open to better names

**Comment by gempa-jabe** [comment]
`stdloc` sound good to me and avoids ambiguities with other locators.

**Comment by gempa-jabe** [comment]
Do you think that the PR is ready to be merged? We can continue at any time in the future improving it.

**Comment by luca-s** [comment]
I have tested it and I am happy with the results,  so you can integrate it if you like.  I still would like to add the confidence ellipsoid to the location, since at the moment there is no uncertainty reported. However, as you said, I can add it in a future PR. 

I don't know if there is someone planning to give it a try before publishing the code in  a SeisComP release. I would be happy if someone could give a critical review at the configuration parameters. Just to be sure it is clear the way they are presented and documented.

**Comment by luca-s** [comment]
Sounds good, thanks.

**Comment by gempa-jabe** [comment]
It seems your tab key has been stolen :wink:

**Comment by luca-s** [comment]
Lol, It's just that I run the code though clang-format with default settings.

**Pull request merged**